### ISSUE TITLE
Possible Virus Infection

### ISSUE THREAD

**Issue by trparky** [dechamps/RudeWindowFixer#1]
I ran the downloadable executable through VirusTotal and it doesn't look good. Four out of 68 virus scanners indicate a possible infection.

https://www.virustotal.com/gui/file/13f30a1f74852c5c675644fbb95bb621bc8af0f58d646a3302b9fe52d0d4a438/detection

**Comment by dechamps** [comment]
TL;DR: these are obvious false positives. I can prove it (see below), and any competent software engineer will be able to verify my proof.

If the RudeWindowFixer executable [from the Releases page](https://github.com/dechamps/RudeWindowFixer/releases/tag/RudeWindowFixer-0.1) was a virus, then that would indicate that *GitHub itself* has been compromised. Which is so incredibly unlikely that it's pretty much guaranteed to be a false positive. Otherwise it would make front page news right about now.

Here's how you can personally verify this:

The executable from the Releases page is *not* one I built myself on my own machine. Instead I let GitHub Actions build it, precisely so I can prove nothing fishy is going on (among other reasons). If you go to this [GitHub Actions](https://github.com/dechamps/RudeWindowFixer/actions/runs/1802934690) run, and you download the `RudeWindowFixer-x86-Release` artefact, you will be able to verify that the executable from GitHub Actions is exactly the same as the one from the GitHub releases page. It is also the exact same file you uploaded to VirusTotal (same SHA-256 hash).

Now, if you go look at the [workflow](https://github.com/dechamps/RudeWindowFixer/blob/ff07fb29ce7e1c77b36bf271e467992fa193c1cc/.github/workflows/continuous-integration.yml) this GitHub Action ran, you will see that it is completely trivial, standard build stuff with no strange dependencies or anything like that. And of course you can also look at the [RudeWindowFixer](https://github.com/dechamps/RudeWindowFixer/blob/ff07fb29ce7e1c77b36bf271e467992fa193c1cc/RudeWindowFixer.c) source code that was built by that workflow, which is so small (80 lines) and trivial that you can easily tell it doesn't do anything suspicious.

What I'm trying to say here is: *you don't have to trust me or the executable I published*. You can verify it yourself. You only have to trust GitHub.

If you're not a software engineer and so don't have the necessary skills/background to appreciate the implications of the above, then please send a link to this page to any competent software engineer and they will be able to verify everything I said.

So given the above, the only way a virus could have slipped in there is if the GitHub Actions runner VM itself (which is hosted by GitHub and not under my control) was infected, and was in turn infecting executables that were being build on that runner. Which, granted, is *technically possible*, but is so mind-bogglingly unlikely that I don't think it's reasonable to entertain that possibility based on a couple of VirusTotal findings.

So in conclusion, I would suggest that these VirusTotal findings are more representative of faulty/overparanoid antivirus software that somehow manages to find a virus in a 80-line C program(??) than an actual threat. (A less cynical explanation would be that some of the RudeWindowFixer qualifies as "unusual" - e.g. the shell hook or the `BroadcastSystemMessage` call - and that makes some antiviruses suspicious. Obviously, just because it's unusual doesn't mean it's harmful. RudeWindowFixer code is unusual because it's solving an unusual problem.)

If you still don't trust the executable I published after reading the above, then ultimately you can always build it yourself. All you need is to install Visual Studio, clone the repo, and hit Build. That's it - you'll then end up with your own executable. Again, no trust required, beyond trusting the source code that you can read for yourself.

**Comment by trparky** [comment]
I didn't know how the executable was built, I thought it was built by you on your personal machine. I'll take a look at the GitHub Actions page.

**Comment by dechamps** [comment]
I'm going to assume you are satisfied with my explanation and close this. Feel free to comment if you have additional questions.

**Comment by ReanimationXP** [comment]
There are now a significant number of AV's that detect the app and quarantine it.  It would be worth an effort to, after you build, upload the main exe to VirusTotal and go through the list of AV's reporting it as bad to get it whitelisted.  Hopefully doing this a few times would "train" their heuristics to stop flagging it.

**Comment by ReanimationXP** [comment]
(Also, great work on the app - thanks for your work and the in-depth explanations on how you fixed it, much appreciated!)
